packages/eslint-plugin-circuit-ui/no-invalid-custom-properties/index.ts

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright 2023, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+const VALID_CUSTOM_PROPERTIES_WITHOUT_PREFIX = [
```

Comment:
I don't like having to keep duplicating these. It means that when we add/change a property, we'll have to change it in many different places and we'll probably end up with inconsistencies sometimes.

I'm wondering if we could somehow have a "source of truth" JSON somewhere, and pull from it to generate:

- the actual property declarations (for now in `BaseStyles`, eventually in CSS)
- documentation
- Storybook themes?
- this thing (regex for lint rules)

Alternatively we could decide that the "source of truth" is the CSS string where variables are declared (in `BaseStyles` for now, simply decoupled from the rest of the base styles). In JS (like in this case) we could parse this CSS string to extract property names as an array. Bit more "hacky" but this would ensure that tools are aligned to the real tokens, and not to some extra JS interface.

Any thoughts here @connor-baer?

Reply:
Also in this specific array, I've had to remove the `--cui` prefix to build my regex. There are probably ways to keep it in by reworking the regex, and I'd like to do that in possible. As a rule of thumb, I'd like to always keep the variables in a single piece

Reply:
I think that’s a good idea, as long as we don’t publicly export the JSON theme. It should remain an implementation detail.

Reply:
I don’t think it’s worth reworking the regexp to include the prefix. It would probably make it much less efficient.

Reply:
Will do in a separate PR since this will touch a number of files and packages 🙂